docs/dev/contributing.rst

```diff
@@ -185,22 +277,101 @@ You can install pre-commit by running:
+    Format YAML files....................................(no files to check)Skipped
+    ruff.....................................................................Failed
+    - hook id: ruff
+    - exit code: 1
+    - files were modified by this hook
 
-If you want you can use pre-commit on it's own to run agains all files for exmaple.
+    hydromt/new_feature.py:1:1: D100 Missing docstring in public module
+    Found 2 errors (1 fixed, 1 remaining).
+
+    black....................................................................Passed
+
+This means that pre-commit has found issues in the code you submitted. In the case of the import it was able to fix it automatically. However `ruff` has also detected that you have not added a docstring for the new feature. You can find this out by running:
+
+.. code-block:: console
+
+  $ ruff .
+
+which will show you the same output:
 
 .. code-block:: console
 
-    $ pre-commit run --all
+  hydromt/new_feature.py:1:1: D100 Missing docstring in public module
+  Found 1 error.
+
+After you've fixed this problem by for example adding the docstring """Implement the cool new feature""" at the top of the new file, you just have to add the new version to the staging area again and re-attempt the commit which should now succeed:
+
+.. code-block:: console
+
+  $ git add hydromt/new_feature.py
+  $ git commit -m "The feature you've all been waiting for."
+  Trim Trailing Whitespace.................................................Passed
+  Fix End of Files.........................................................Passed
+  Check Yaml...........................................(no files to check)Skipped
+  Check for added large files..............................................Passed
+  Check python ast.........................................................Passed
+  Check JSON...........................................(no files to check)Skipped
+  Debug Statements (Python)................................................Passed
+  Mixed line ending........................................................Passed
+  Format YAML files....................................(no files to check)Skipped
+  ruff.....................................................................Passed
+  black....................................................................Passed
+  [linting a5e9b683] The feature you've all been waiting for.
+   1 file changed, 4 insertions(+)
+   create mode 100644 hydromt/new_feature.py
+
+Now you can push your commit as normal.
+
+From time to time you might see comments like these:
+
+.. code-block:: python
+
+  import rioxarray # noqa: F401
+
+The `noqa` is instructing the linters to igore the specified rule for the line in question. Whenever possible, we try to avoid using these but it's not allways possible. The full list of rules can be found here: `Ruff Rules Seciont <https://beta.ruff.rs/docs/rules/>`__ Some common ones are:
```

Comment:
```suggestion
The `noqa` is instructing the linters to ignore the specified rule for the line in question. Whenever possible, we try to avoid using these but it's not always possible. The full list of rules can be found here: `Ruff Rules Section <https://beta.ruff.rs/docs/rules/>`_ Some common ones are:
```